price-estimator/src/main.rs

```diff
@@ -158,7 +158,14 @@ fn main() {
         options.orderbook_update_interval,
     ));
 
-    let filter = filter::all(orderbook, token_info).with(warp::log("price_estimator"));
+    // We add the allow origin header so that requests from the interactive openapi documentation
+    // go through to locally running instance.
+    let filter = filter::all(orderbook, token_info)
+        .with(warp::log("price_estimator"))
+        .with(warp::reply::with::header(
+            "Access-Control-Allow-Origin",
+            "*",
+        ));
```

Comment:
Maybe this should only be enabled in debug builds, just in case. I expect that nginx should set these headers and not the `price-estimator` in production.

Reply:
We also want the openapi to work with the staging and production and nodes and either way this has no security impact so it makes sense to me that this is always enabled.

Reply:
> We also want the openapi to work with the staging and production

This is already the case because of the nginx configuration:
```
curl -i -X OPTIONS https://price-estimate-mainnet.dev.gnosisdev.com/                                                                  
HTTP/1.1 204 No Content
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization
Access-Control-Allow-Methods: OPTIONS, GET, HEAD
Access-Control-Allow-Origin: *
Access-Control-Max-Age: 1728000
Content-Length: 0
Date: Thu, 13 Aug 2020 18:45:49 GMT
Connection: keep-alive
```

> this has no security impact 

This isn't strictly true, it enables browsers to make arbitrary requests to the service. Its much easier to get a whole bunch of browsers at different IPs to make HTTP requests than a whole bunch of applications. I do agree that this is not a realistic threat.

> that this is always enabled.

My main argument is that should be a deployment configuration and not a hard-coded thing. But either way, I don't think its super important in this case.